### PR TITLE
Implement a built-in function for String count

### LINKED ISSUE #6827: Add `strings.count(string, substring)` built-in function
For cases where you need to count the occurances of a substring in some text, it would be nice if instead of this:
```rego
n := count(indexof_n(string, substring))
```
One could do this:
```rego
n := strings.count(string, substring)
```

"Nice" in that it more clearly communicates intent, would be easier to find in documentation, and likely can be made more performant.

### PR DESCRIPTION
**Pull request by Manish-Giri**
Fixes #6827

### Why the changes in this PR are needed?

These changes would be an enhancement over first having to generate the indices, then again a count of the indices.

### What are the changes in this PR?

Adding support for a `strings.Count` built-in function, that would return the count of occurrences directly. Includes tests.

### CODE REVIEW

File: ast/builtins.go [hunk 1]
```diff
@@ -1109,6 +1110,19 @@ var Contains = &Builtin{
 	Categories: stringsCat,
 }
 
+var StringCount = &Builtin{
+	Name:        "strings.count",
+	Description: "Returns the number of non-overlapping instances of a substring in a string.",
+	Decl: types.NewFunction(
+		types.Args(
+			types.Named("cheese", types.S).Description("string to search in"),
+			types.Named("e", types.S).Description("substring to look for"),
```
Comment (srenatus) [inline ast/builtins.go:1119]: Sorry I didn't notice this before: can we please change the names, like
```suggestion
			types.Named("search", types.S).Description("string to search in"),
			types.Named("substring", types.S).Description("substring to look for"),
```

**Review by srenatus — COMMENTED** [review]
One last nitpick. Thanks for implementing this 🥳

File: ast/builtins.go (hunk 1 above)
Comment (Manish-Giri) [reply ast/builtins.go]: @srenatus Thank you, Updated accordingly!

**Review by srenatus — DISMISSED** [review]

**Comment by Manish-Giri** [comment]
Looks like there is a Yml lint issue, should I submit the fix quickly?

File: builtin_metadata.json [hunk 2]
```diff
@@ -17382,6 +17383,31 @@
...
+      {
+        "description": "string to search in",
+        "name": "cheese",
+        "type": "string"
+      },
+      {
+        "description": "substring to look for",
+        "name": "e",
+        "type": "string"
+      }
```
Comment (ashutosh-narkar) [inline builtin_metadata.json:17397]: @Manish-Giri the names here are probably from a previous iteration. I think a `make generate` should take care of this.

Comment (Manish-Giri) [reply builtin_metadata.json]: @ashutosh-narkar Updated

**Review by ashutosh-narkar — APPROVED** [review]

**Pull request merged**